pyvista/core/filters/uniform_grid.py

```diff
@@ -47,8 +42,12 @@ def gaussian_smooth(
         alg.SetInputDataObject(self)
         if scalars is None:
             field, scalars = self.active_scalars_info
+            if field.value == 1:
+                raise ValueError('If `scalars` not given, active scalars must be point array.')
         else:
-            field = self.get_array_association(scalars, preference=preference)
+            field = self.get_array_association(scalars, preference='point')
+            if field.value == 1:
+                raise ValueError('Can only process point data, given `scalars` are cell data.')
```

Comment:
We could recommend to the user to use the `cell_data_to_point_data` filter in the error message.

Reply:
Yeah I wasn't sure how verbose we wanted this to be. Maybe this could be added as a new PyVista-specific error in core/errors.py and then added wherever relevant (including for the other filters you mentioned). Probably a subject for another issue/PR.

Reply:
I'm okay leaving this for a more general solution across the code if you want to keep this fix PR small.

Reply:
Yes, I don't have the time or the familiarity with pyvista's codebase to expand the scope of this PR beyond the uniformgrid filters. I'm fine with changing the error messages I've used here to be more informative but it might be better for someone else to make the decision of what feedback the user should get in this situation (trying to use cell data in a filter that doesn't support it) and then apply that consistently across the board.